unit-tests/src/test/scala/java/net/UdpSocketTest.scala

```diff
@@ -0,0 +1,139 @@
+package java.net
+
+import scalanative.posix.arpa.inet._
+import scalanative.posix.fcntl
+import scalanative.posix.fcntl.{F_SETFL, O_NONBLOCK}
+import scalanative.posix.netinet.in._
+import scalanative.posix.netinet.inOps._
+import scalanative.posix.sys.socket._
+import scalanative.posix.sys.socketOps._
+import scalanative.posix.unistd.close
+import scalanative.unsafe._
+import scalanative.unsigned._
+
+import org.junit.Test
+import org.junit.Assert._
+
+class UdpSocketTest {
+  // All tests in this class assume that an IPv4 network is up & running.
+
+  @Test def sendtoRecvfrom(): Unit = Zone { implicit z =>
+    val localhost         = c"127.0.0.1"
+    val localhostInetAddr = inet_addr(localhost)
+
+    val inSocket = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)
+    assertNotEquals("socket", -1, inSocket)
+
+    try {
+      val inAddr      = alloc[sockaddr]
+      val inAddrInPtr = inAddr.asInstanceOf[Ptr[sockaddr_in]]
+
+      inAddrInPtr.sin_family = AF_INET.toUShort
+      inAddrInPtr.sin_addr.s_addr = localhostInetAddr
+      // inAddrInPtr.sin_port is already the desired 0; "find a free port".
+
+      val fcntlStatus = fcntl.fcntl(inSocket, F_SETFL, O_NONBLOCK)
+      assertNotEquals("fcntl", -1, fcntlStatus)
+
+      // Get port for sendto() to use.
+      val bindStatus = bind(inSocket, inAddr, sizeof[sockaddr].toUInt)
+      assertNotEquals("bind", -1, bindStatus)
+
+      val inAddrInfo = alloc[sockaddr]
+      val gsnAddrLen = alloc[socklen_t]
+      !gsnAddrLen = sizeof[sockaddr].toUInt
+
+      val gsnStatus = getsockname(inSocket, inAddrInfo, gsnAddrLen)
+      assertNotEquals("getsockname", -1, gsnStatus)
+
+      // Now use port.
+      val outSocket = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP)
+      assertNotEquals("socket", -1, outSocket)
```

Comment:
Also here we have the same undistinguishable error message for two assertions  `inSocket`/`outSocket`

Reply:
Fixed, thank you.